dbt_subprojects/daily_spellbook/models/_sector/addresses/addresses_info.sql

```diff
@@ -31,240 +30,218 @@
+    , map_union(map_from_entries(array[
+            (blockchain, chain_stats)
+            ])) AS chain_stats
+    , MAX(last_seen) AS last_seen
+    , MAX(last_seen_block) AS last_seen_block
+    FROM (
+        {% for addresses_model in addresses_models %}
+        SELECT '{{ addresses_model[0] }}' AS blockchain
+        , address
+        , executed_tx_count
+        , max_nonce
+        , is_smart_contract
+        , namespace
+        , name
+        , first_funded_by
+        , first_funded_by_block_time
+        , tokens_received_count
+        , tokens_received_tx_count
+        , tokens_sent_count
+        , tokens_sent_tx_count
+        , first_transfer_block_time
+        , last_transfer_block_time
+        , first_received_block_number
+        , last_received_block_number
+        , first_sent_block_number
+        , last_sent_block_number
+        , received_volume_usd
+        , sent_volume_usd
+        , first_tx_block_time
+        , last_tx_block_time
+        , first_tx_block_number
+        , last_tx_block_number
+        , map_from_entries(array[
+            ('last_seen', CAST(last_seen AS varchar))
+            , ('executed_tx_count', CAST(executed_tx_count AS varchar))
+            , ('tokens_sent_count', CAST(tokens_sent_count AS varchar))
+            , ('tokens_received_count', CAST(tokens_received_count AS varchar))
+            ]) AS chain_stats
+        , last_seen
+        , last_seen_block
+        FROM {{ addresses_model[1] }}
+        {% if not loop.last %}
+        UNION ALL
+        {% endif %}
+        {% endfor %}
+    )
+    GROUP BY address
+    )
+
+SELECT address
+, blockchains
+, executed_tx_count
+, max_nonce
+, max_nonce_blockchain
+, is_smart_contract
+, smart_contract_blockchains
+, namespace
+, name
+, first_funded_by
+, first_funded_blockchain
+, first_funded_by_block_time
+, tokens_received_count
+, tokens_received_tx_count
+, tokens_sent_count
+, tokens_sent_tx_count
+, first_transfer_block_time
+, last_transfer_block_time
+, first_received_block_number
+, last_received_block_number
+, first_sent_block_number
+, last_sent_block_number
+, sent_volume_usd
+, received_volume_usd
+, first_tx_block_time
+, last_tx_block_time
+, chain_stats
+, last_seen
+, last_seen_block
+FROM data
+
+
+
+{% else %}
+
+
+
+WITH to_update AS (
+    SELECT DISTINCT am.address
+    FROM (
+        {% for addresses_model in addresses_models %}
+        (SELECT address
+        , last_seen_block
+        , '{{ addresses_model[0] }}' AS blockchain
+        FROM {{ addresses_model[1] }}
+        WHERE {{incremental_predicate('last_seen')}}
+        )
+        {% if not loop.last %}
+        UNION ALL
+        {% endif %}
+        {% endfor %}
+        ) am
+    LEFT JOIN {{ this }} t ON am.address = t.address
+    WHERE t.address IS NULL
+        OR ((contains(t.blockchains, am.blockchain) = FALSE))
+        OR (CAST(t.chain_stats[am.blockchain]['last_seen_block'] AS bigint) <= am.last_seen_block)
```

Comment:
we still rely on `last_seen_block` here, that should be adapted to use `last_seen`

Reply:
fixed